docs/GCE.md

```diff
@@ -1,63 +1,110 @@
 gcloud auth login
 ```
-
 Also, create a project through the GCE console and set that as the default for gcloud.
 ```
 gcloud config set project PROJECT
 ```
-
 And set a default zone
 ```
 gcloud config set compute/zone us-central1-a
 ```
+## Setting up GCE networking
+GCE blocks traffic between hosts by default, which prevents Calico traffic from flowing.  Run the following command to allow traffic to flow between containers:
+```
+gcloud compute firewall-rules create calico-ipip --allow 4 --network "default" --source-ranges "10.240.0.0/16"
+```
+You can verify the rule with this command:
+```
+gcloud compute firewall-rules list
+```
 
 ## Spinning up the VMs
-You must make sure that the "IP forwarding" flag is set when you first configure the compute nodes (under advanced options in the web developer console, or by specifying the `--can-ip-forward` flag when creating the hosts from the command line).
+etcd needs to be running on the Calico hosts.  The easiest way to bootstrap etcd is with a discovery URL.  Choose an etcd cluster size that is equal to or less than the number of Calico nodes (an odd number in the range 3-9 works well).  We'll use 3 for the size of the etcd cluster and the Calico nodes in the instructions below.  Use `curl` to get a fresh discovery URL (replace size=3 with your cluster size if desired):
+```
+curl https://discovery.etcd.io/new?size=3
+```
+You need to grab a fresh URL each time you bootstrap a cluster.
 
-Run the master with
+Create a file `cloud-config.yaml` with the following contents; **replace `<discovery URL>` with the URL retrieved above**:
 ```
-gcloud compute instances create master --image https://www.googleapis.com/compute/v1/projects/coreos-cloud/global/images/coreos-alpha-618-0-0-v20150312 --machine-type g1-small --metadata-from-file user-data=cloud-config-master.yaml --can-ip-forward
+#cloud-config
+coreos:
+  update:
+    reboot-strategy: etcd-lock
+  etcd2:
+    name: $private_ipv4
+    discovery: <discovery URL>
+    advertise-client-urls: http://$private_ipv4:2379
+    initial-advertise-peer-urls: http://$private_ipv4:2380
+    listen-client-urls: http://0.0.0.0:2379,http://0.0.0.0:4001
+    listen-peer-urls: http://$private_ipv4:2380,http://$private_ipv4:7001
+  units:
+    - name: etcd2.service
+      command: start
+
 ```
+Note: we disable automated reboots for this demo to avoid interrupting the instructions.
 
-And the compute nodes with
+Then create the cluster with the following command ("calico-1 calico-2 calico-3" is the list of nodes to create):
 ```
-gcloud compute instances create core1 core2 --image https://www.googleapis.com/compute/v1/projects/coreos-cloud/global/images/coreos-alpha-618-0-0-v20150312 --machine-type g1-small --metadata-from-file user-data=cloud-config.yaml --can-ip-forward
+gcloud compute instances create \
+  calico-1 calico-2 calico-3 \
+  --image https://www.googleapis.com/compute/v1/projects/coreos-cloud/global/images/coreos-alpha-709-0-0-v20150611 \
```

Comment:
This should work instead if we want to simplify slightly (though it is now two options rather than one):
--image-project coreos-cloud
--image coreos-alpha-709-0-0-v20150611